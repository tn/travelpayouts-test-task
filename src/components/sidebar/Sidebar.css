@import '../../Config.css';

.sidebar {
    width: var(--sidebarWidth);
    height: 100vh;
    padding: 20px 0;
    background-color: var(--sidebarBgColor);
}
