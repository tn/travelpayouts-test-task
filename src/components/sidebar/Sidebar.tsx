import * as React from 'react'
import './Sidebar.css'
import Logo from '../logo/Logo'
import Nav from '../nav/Nav'

const Sidebar: React.SFC = ({ children }) => (
  <aside className='sidebar'>
    <a href='/'><Logo /></a>
    <Nav />
  </aside>
)

export default Sidebar
