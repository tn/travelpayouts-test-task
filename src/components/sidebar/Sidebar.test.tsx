import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Sidebar from './Sidebar'

configure({ adapter: new Adapter() })

describe('Sidebar', () => {
  it('should be rendered', () => {
    const wrapper = renderer.create(<Sidebar />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
