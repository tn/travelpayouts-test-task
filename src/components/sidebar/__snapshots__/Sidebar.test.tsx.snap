// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should be rendered 1`] = `
<aside
  className="sidebar"
>
  <a
    href="/"
  >
    <span
      className="logo"
    />
  </a>
  <nav
    className="nav"
  >
    <ul
      className="nav__list"
    >
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link nav__link--active"
          href="#"
        />
      </li>
      <li
        className="nav__item"
      >
        <a
          className="nav__link "
          href="#"
        />
      </li>
    </ul>
  </nav>
</aside>
`;
